ui/src/Queues/QueuesTable.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import TableContainer from "@material-ui/core/TableContainer";
+import Paper from "@material-ui/core/Paper";
+import Table from "@material-ui/core/Table";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import {TableBody} from "@material-ui/core";
+import "../styles/queue.css";
+import QueueTableRow from "./QueueRow";
+import RefreshQueuesData from "./RefreshQueuesData";
+import useEffectOnce from "./UseEffectOnce";
+
+const QueuesTable: React.FC = () => {
+    const queuesOverallData = RefreshQueuesData.useRefreshQueueData()
```

Comment:
I suggest to move that logic into a separate 
```
const useBlah = () => {
    const queuesOverallData = RefreshQueuesData.useRefreshQueueData()

      useEffectOnce(() => {
         const fetchInitialStatistics = async () => {
             const initialStatistics = await queuesOverallData.obtainInitialStatistics()
             queuesOverallData.setQueuesOverallData((prevState) => {
                 if (prevState.length === 0) {
                     return initialStatistics
                 } else {
                     return prevState;
                 }
             })
         }
         fetchInitialStatistics();
     });
    return queuesOverallData.queuesOverallData;
}
```

and here u would use it as 
```
const queuesOverallData = useBlah();
```
As I can see the rest of this component cares only about `queuesOverallData`

Reply:
I have changed a little bit this fragment and right now it is just a one liner as You have proposed